src/index/base.cpp

```diff
@@ -160,7 +160,7 @@ void BaseIndex::ThreadSync()
                 pindex = pindex_next;
             }
 
-            int64_t current_time = GetTime();
+            auto current_time{GetTime<std::chrono::seconds>()};
```

Comment:
given that the clock is used to schedule things, an alternative would be to change this to a steady clock. This will behave better if the system clock is adjusted. However std::chrono::steady_clock is not mockable, which I think is fine in this context here.

```suggestion
            auto current_time{std::chrono::steady_clock::now()};
```

Reply:
`steady_clock::now()` returns `steady_clock::time_point` and this results in compilation error in the next line (`last_log_time + SYNC_LOG_INTERVAL < current_time` - `no operator "<" matches these operands`).

With `duration_cast`, it seems to work.

```diff
- auto current_time{GetTime<std::chrono::seconds>()};
+ auto current_time{std::chrono::duration_cast<std::chrono::seconds>(std::chrono::steady_clock::now().time_since_epoch())};
```

Reply:
last_log_time should be of the same type.

`std::chrono::stead_clock::duration`

Reply:
I'm not very familiar with the `std::chrono::steady_clock` interface but I think https://github.com/bitcoin/bitcoin/pull/25079/commits/74a70ebea07ccbf5c7402f932b1880c499dd3ac0 does the trick.